### ISSUE TITLE
INI file bloats to 100 MB

### ISSUE THREAD

**Issue by dca00** [texstudio-org/texstudio#1232]
### Environment
- TeXstudio: 2.12.6
- Qt: can't see, it does not start
- OS: Linux fedora 26
- TeX distribution: texlive 2018

### Expected behavior

Texstudio should not use up 2.5 GB of RAM.

### Actual behavior

This is not really a bug report but a plea for help. I am locked up on this Fedora 26 environment that is vendor-maintained and, therefore, cannot be updated. So please do not close this as "you have to upgrade". I am not asking you to fix the issue - only to explain how to get Texstudio up and running again.

I noticed that Texstudio takes 2.5 GB of RAM. I started looking through the .config/texstudio and found that its INI file is 103 MB in size. The bulk of it consists of 1.5 million instances of

keysetting\1529155\id=main/edit/redo~0
keysetting\1529155\key=Ctrl+Y

Above is the last instance of this pattern.

There is a line right after the first 10 of such groups:
keysetting\size=1529155

I tried deleting all of these groups except for the first one and changing the number above to 1.
After that, Texstudio did not start. It starts, waits for a few seconds w/o showing its window, and quits.
I then restored texstudio.ini from the backup of that 103 MB copy. It still does not start - same behaviour.

I downloaded texstudio RPM for my version, in a hope that there is a vanilla INI inside. I could not find it.

Please assist in obtaining a working, vanilla INI file.

### How to reproduce

I don't know.

**Comment by dca00** [comment]
I just noticed that the same issue is happening in the .texprofile files as well. They have the same repetition of the redo commands.

**Comment by dca00** [comment]
This is cool now! I found that despite me quitting Texstudio on the last session, it remained in the background, running without a window. I killed the process, and now it starts normally, after I erased all entries and reset \size key to 0

**Comment by dca00** [comment]
May I make a suggestion?
These pesky keysettings are interspersed with Macros, thus removing them is very tricky and time consuming, in the profile files.
Could you please put the macros on a side, separately from keysettings? It only makes sense. Needless to say, you should get to the bottom of the issue with exponential growth of the keysettings keys.

**Comment by dca00** [comment]
Wow!
I just started Texstudio with the INI file that I stripped of keysettings, and they are back again, all 122495 of them.
There is a serious problem in their generation, so it seems. They procreate, like rabbits.

**Comment by sunderme** [comment]
seems odd.
You are aware that your version of txs is rather old and we are not debugging that ?
I would strongly suggest to install the current version (3.0.0).

**Comment by dca00** [comment]
I am, as stated in the preamble of this issue.
Thanks for your suggestion! It did occur to me too. But like I said, I am in a locked, vendor-supported environment, that will not change.
It is still worth looking into the code responsible for regular updates of the INI files and finding out why it multiplies the keysettings key so many times. Eventually, Texstudio grinds to a halt, takes forever to load up and to close, and as evident from my last experience, becomes a zombie process. And I am lucky to be on a device that has 16 GB of physical RAM. You can imagine someone on a 4 or 8 GB machine!

**Comment by sunderme** [comment]
maybe appimage is an alternative ? No root rights necessary, runs as a single file.
As old as your version is, there has never been a complaint like yours. And unless the bug is shown to be present in the current version, no bug hunting ...

**Comment by sunderme** [comment]
to answer your question, remove the ini . txs generates a pristine one if it does not find an existing ini.

**Comment by dca00** [comment]
You are not reading the thread. I've already stated how this had to be fixed.

**Comment by dca00** [comment]
The bloat happens when a profile is being switched.
Loading Texstudio with a clean INI does not lead to its bloating, but as soon as I switch a profile, it bloats.
I just loaded a profile (that had been cleaned from keysettings completely) and exactly 122495 new keysettings (empty, this time) had been added to the INI.
It is perfectly normal that your other users never reported this issue. They likely never load multiple profiles, and many likely do not even have profiles.
I am a multilingual technical writer, and I systematically reload profiles within one session, to work on different language versions of the same document, so I am a victim to a bug that no one else notices.
